src/scheduler/fairshare.cpp

```diff
@@ -919,13 +917,23 @@ fairshare_head::fairshare_head()
  * @param[in]	ofhead	-	fairshare_head to dup
  *
  * @return	duplicated fairshare_head
- * @retval	NULL	: fail
+ * @retval	NULL	: fail 
  */
 fairshare_head::fairshare_head(fairshare_head& ofhead)
 {
 	last_decay = ofhead.last_decay;
 	root = dup_fairshare_tree(ofhead.root, NULL);
+}
 
+/**
+ * @brief copy assignment operator for fairshare_head
+ */
+fairshare_head fairshare_head::operator=(fairshare_head& ofhead)
```

Comment:
Shouldn't this return a reference rather than the object?

Reply:
Indeed it should.  Thanks for catching this.